### ISSUE TITLE
editing scss files doesn't seem to update styles on page

### ISSUE THREAD

**Issue by mbaxter91288** [google/web-starter-kit#326]
hello all, not sure if it's me missing something or what, but I have cloned the repo and run 'npm install' from the directory with no issues. However, once running 'gulp serve', when I go to change any scss files gulp seems to run the 'styles' job like it should and the page reloads, but none of the styles change. The gulp file seems to update the '.tmp/styles/*' files correctly, but the reload doesn't then use these files.

Simple example, deleting the entire contents of the 'app/styles/components/components.scss' file and saving seems to run all the gulp tasks and reloads the browser, but all styling stays the same.

Any help would be much appreciated, I'm sure it's something I am doing stupidly wrong.

Many thanks in advance.
Mat

**Comment by komplexb** [comment]
I'm having this exact problem with the latest version I downloaded here: https://github.com/google/web-starter-kit/releases/tag/v0.5.4

**Comment by sonicbobcat85** [comment]
I'm experiencing the same issue as well, using the latest release of GWST.

**Comment by Bluematter** [comment]
This issue has come back. If I make a change to a partial `app/styles/components/_components/_button.scss` as an example, `.pipe($.changed('.tmp/styles', {extension: '.css'}))` won't pipe the changed partials because it's not a part of the src, only changes to the src files `/app/styles/main.scss` or `app/styles/components/components.scss` will compile a new temporary css file?

The above is quite obvious so I assume that it is `sourcemaps` job, to handle that (don't know what it does)? If so maybe there is an issue with that?

What I have been doing is just leaving `main.scss` open and hitting space then save, I finally am irritated with that method lol

**Comment by Bluematter** [comment]
I can confirm something is wrong with my gulp.js file, after copying the master directly over it works, I added browserify tasks and some other ones, will figure out what is causing it.

**Comment by joedajigalo** [comment]
Confirmed issue also for me. 

Adding styles to main.scss reloads correctly with the changes displaying but if a scss (ie. _partial.scss) file is imported to main.scss and that file has styles added, the browser reloads but the change are not displaying. 

However, if you do make changes to the _partial.scss and than write new styles to main.scss, than the styling from _partial.scss displays.

**Comment by addyosmani** [comment]
> I can confirm something is wrong with my gulp.js file, after copying the master directly over it works

Can you confirm that master is working fine for you?

@ joedajigalo does master continue to persist with the issues in your comment too?

**Comment by joedajigalo** [comment]
@addyosmami yes, just pulled from master last night.

**Comment by joedajigalo** [comment]
I believe that the issue is since the .scss files are @imported into main.scss, `gulp-changed` cannot see the updated .scss files?

**Comment by Bluematter** [comment]
After modifying the gulp.js file for browserify somewhere down the line the modifications caused this issue, but I do confirm that after copying the gulp.js file in the master branch it works smooth. I wish I could pinpoint the exact issue but I deleted the faulty code. I re added browserify and it still works great. 

I do wish we could work on a browserify branch, because what's better than the ability to keep your js modular, merge all your js dependencies into one minified script? Also when you hook it up to watchify and browser reload, you can develop some pretty awesome things rapidly.

I wish I knew why you guys who are copying from master are having issues, did you (painfully) try to re install your node modules? Maybe completely deleting them is overkill, but a simple `npm install` might fix things? Did you double check your package.json? Are you cloning a fresh master, or just merging components?

**Comment by Bluematter** [comment]
@joedajigalo since you're doing a pull maybe there is still something wrong locally, because I don't think a pull will fix that. If you made modifications to your gulp.js I would comment it out, put it to the side, copy the master's gulp.js and try that. If that does not work, re install node_modules, try again.

@addyosmani like I mentioned above master works fine, I even did a fresh clone, and that definitely works, I think the people with issues have to have something wrong locally, and I don't think a pull is going to fix it until they identify the issue locally, or get a fresh copy and manually merge their components etc, also I am sure there is also a better way with git, there is always something cool in git I don't know, but I am willing to bet the issue is from a manual change and lives locally.

**Comment by sonicbobcat85** [comment]
Now it looks like the problem is happening on other GWST projects on the same machine that used to run fine.  I haven't made any changes to gulp; maybe node or sass?

**Comment by Bluematter** [comment]
So as of late I have pulled WSK for many different projects, I notice when I delete all the default modules to add my own that is when the issue starts to occur.

I am not sure how changed works but I think it caches the original modules/components but the new ones are not recognized?

**Comment by addyosmani** [comment]
I landed some changes to our build process two days ago which should help with this. Could you pull from master and try again?

**Comment by Bluematter** [comment]
It looks like your solution solved it, the only issue is the styles take about 1mb to compile. I will look further into why, and see if perhaps it's something on my end. I think it's primarily with `gulp changed` because I am also having issues with changed handlebars templates, I probably am missing something . The idea behind `gulp changed` is fantastic, but it's causing issues for me, unfortunately.

**Comment by Bluematter** [comment]
`gulp-newer` works like a charm. I know sindresorhus created `gulp-changed` but the former works well since its many:1 ie many partials to one build sheet.

**Comment by wearenifty** [comment]
I had the exact same issue but solved it by adding an additional task to clean out the css before re-compiling. This forces a re-compile of all SASS files which may be inefficient for larger projects but for my purposes works perfectly:

```
gulp.task('cleanCss', del.bind(null, ['.tmp/styles', 'dist/styles'], {dot: true}));

// Watch files for changes & reload
gulp.task('serve', ['styles', 'jade'], function () {

    browserSync({
        notify: true,
        // Customize the BrowserSync console logging prefix
        logPrefix: 'WSK',
        // Run as an https by uncommenting 'https: true'
        // Note: this uses an unsigned certificate which on first access
        //       will present a certificate warning in the browser.
        // https: true,
        server: ['.tmp', 'app']
    });

    gulp.watch(['app/jade/**/*.jade'], ['jade', reload]);
    gulp.watch(['app/styles/**/*.{scss,css}'], ['cleanCss', 'styles', reload]);
    gulp.watch(['app/js/**/*.js'], ['jshint'], reload);
    gulp.watch(['app/images/**/*'], reload);
});

```

**Comment by PeanutButterCo** [comment]
Have the same issue in release 0.5.4. Like @Bluematter says, removing , {extension: '.css'} is a quick fix. ¿Is there any problem with this solution?

**Comment by nicolasgarnier** [comment]
Seems like the issue is caused by `gulp-changed` not supporting many:1 file transformations.
I'll investigate switching to `gulp-newer`.

@PeanutButterCo I guess removing `, {extension: '.css'}` is like removing the whole line. You'll loose the optimization of only processing newer/modified files because the file names never matches (the option is here to tell the plugin that the output of main.scss is main.css)

**Comment by PeanutButterCo** [comment]
Styles are still not updated when editing any scss file other than main.scss.

**Comment by nicolasgarnier** [comment]
@PeanutButterCo is right.
Sorry for not having investigated this better initially but basically it seems that we won't be able to use gulp-newer or gulp-changed for this task unless we are sure to have only one output file (which our gulpfile does not enforce atm). To sum it up gulp-newer does support 1:1 (like gulp-changed) and many:1 but not many:many. This is a problem in the case of scss because, for instance, a file could be imported from multiple other top level scss files...

Basically we have to remove filtering with gulp-changed entirely - at least for scss files - here unless we implement a filtering logic that's not trivial at all (basically, for scss, we have to construct the import dependency tree). What I'll do is use this rule:

`.pipe($.newer('.tmp/styles'))`

This will effectively work if you are writing CSS files but for SCSS files the comparison will always fail so they will always get processed by the task.

**Comment by Kangomnawe** [comment]
I am having same problem with my style.css on local environment setup with VCCW.
I used underscored start-up-theme from scratch building new word-press theme but unfortunately there is no changes taking place in css while making changes on scss.

**Comment by effizo** [comment]
>"there is no changes taking place in css while making changes on scss." I have the same issue. Please have you been able to resolve this issue? Thank you.
